src/search.cpp

```diff
@@ -615,21 +615,24 @@ namespace {
     if (!rootNode)
         (ss+2)->statScore = 0;
 
-    // Step 4. Transposition table lookup. We don't want the score of a partial
-    // search to overwrite a previous full search TT value, so we use a different
-    // position key in case of an excluded move.
+    // Step 4. Transposition table lookup.
     excludedMove = ss->excludedMove;
-    posKey = excludedMove == MOVE_NONE ? pos.key() : pos.key() ^ make_key(excludedMove);
+    bool preExcludedTtHit = ss->ttHit; // to guard against TT race conditions
```

Comment:
I must admit I don't understand what this has to do with race conditions?

Reply:
My thougths were following:
If  preExcludedTtHit  == false then after calling tte = TT.probe(posKey, ss->ttHit) ss->ttHit will also be false when running singlethreaded. For single thread the variable preExcludedTtHit  is useless.
But under race conditions (multiple threads) it is possible that preExcludedTtHit  == false and then in the SE node with excludedMove suddenly tte = TT.probe(posKey, ss->ttHit) produces a hit.

However I just realized that even if this happens, it doesn't influence the logic at all.
I feared that in this case we inadvertently take ss->static-eval from TT but it's not the case,
so we can simplify away this variable.

Reply:
Forget my last sentence, the variable is needed for following scenario:
multiple threads: preExcludedTtHit == true and then in the SE node with excludedMove suddenly tte = TT.probe(posKey, ss->ttHit) fails to produce a hit. ss->static-eval then contains the value taken from TT.
If we write
````
    else if (excludedMove)
    {
       if (ss->ttHit)
           ss->staticEval = eval = evaluate(pos, &complexity);
       else 
          eval = ss->staticEval;
````
instead to
````
    else if (excludedMove)
    {
       if (ss-> preExcludedTtHit )
           ss->staticEval = eval = evaluate(pos, &complexity);
       else 
          eval = ss->staticEval;
````
then we take eval = ss->staticEval which comes from TT which we dont want.

Reply:
~~> ss->ttHit will also be false when running singlethreaded. 

is this so? just because it was not a tt hit in the calling search doesn't mean it wasn't saved in line 750/764.

> For single thread the variable preExcludedTtHit is useless.

I dont think so?

in other words:

1) outer search at the pos

2) outer search either gets a tthit, or else gets a fresh eval and `save`s the eval with bound/depth/value/move NONE.

3) outer search passes singular ex~~

wait hang on. by definition, we only fall into singular extension block when `move == ttMove`, which implies a ttHit, except:

>     ttMove =  rootNode ? thisThread->rootMoves[thisThread->pvIdx].pv[0]
            : ss->ttHit    ? tte->move() : MOVE_NONE;

so by definition `preExcludedTtHit` is always true, except at the rootNode.

so we only ever re-use the rootNode static eval, which, there's no way that's the source of the eval gain.

so what the hell gained?? is it merely excluding the complexity when excluded? excluding the tt cutoffs? excluding the static `save`?

Reply:
Ups, you are right. Line 740 don't even get executed once with a larger bench run.
I completely missed that.
I guess the +1 Elo then comes from avoiding these false positives i mentioned under point 1
At this point  I'm asking @vondele if I should close this PR and restart from scratch,
or if it's fine to simplify away the unneeded stuff without further testing...

Reply:
yes, I think it is best to close the PR and retest a variant that takes all the discussion into account, i.e. that would correspond to the version to be committed.

Reply:
well we can certainly rerun a functionally-identical patch with the nonfunctional variable removed, and in this case i say go straight to LTC